src/main/java/org/broadinstitute/consent/http/resources/DataAccessRequestResource.java

```diff
@@ -153,10 +154,27 @@ public Response describeManageDataAccessRequests(@Auth AuthUser authUser) {
     @GET
     @Produces("application/json")
     @Path("/manage/v2")
-    @RolesAllowed({ADMIN, CHAIRPERSON, MEMBER})
-    public Response describeManageDataAccessRequestsV2(@Auth AuthUser authUser) {
-        List<DataAccessRequestManage> dars = dataAccessRequestService.describeDataAccessRequestManageV2(authUser);
-        return Response.ok().entity(dars).build();
+    @RolesAllowed({ADMIN, CHAIRPERSON, MEMBER, SIGNINGOFFICIAL})
+    public Response describeManageDataAccessRequestsV2(@Auth AuthUser authUser, @QueryParam("roleName") Optional<String> roleName) {
+        try {
+            User user = userService.findUserByEmail(authUser.getName());
+            String roleNameValue = roleName.orElse(null);
+            if (Objects.nonNull(roleNameValue)) {
+                //if the roleName is SO and the user does not have that role throw an exception
+                if (roleNameValue.equals(UserRoles.SIGNINGOFFICIAL.getRoleName())) {
+                    if (!user.hasUserRole(UserRoles.SIGNINGOFFICIAL)) {
+                        throw new NotFoundException("User: " + user.getDisplayName() + ", " + " does not have Signing Official role.");
+                    }
+                //if there is a roleName but it is not SO then throw an exception
+                } else {
+                    throw new BadRequestException("Invalid role name: " + roleNameValue);
+                }
```

Comment:
If I enter "Member", that's a valid role name. I think the message should say, "Signing Official Role is the only role supported at this time" or something like that. If I enter "French Fries", that's an **_invalid_** role name (see comment above) so we really need two levels of validation here.